### PR TITLE
I reviewed start_opencl_environment() as it is now

### PR DESCRIPTION
**Pull request by claudioandre-br**
Just documenting my findings here

### CODE REVIEW

File: src/opencl_common.c [hunk 1]
```diff
@@ -114,10 +114,10 @@ typedef struct {
 	cl_platform_id platform;
 	int num_devices;
 } cl_platform;
-static cl_platform platforms[MAX_PLATFORMS];
+static cl_platform platforms[MAX_PLATFORMS + 1];
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Allow MAX_* plus the ending `NULL`

File: src/opencl_common.c [hunk 2]
```diff
@@ -114,10 +114,10 @@ typedef struct {
...
 
 
-cl_device_id devices[MAX_GPU_DEVICES];
+cl_device_id devices[MAX_GPU_DEVICES + 1];
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Idem

File: src/opencl_common.c [hunk 3]
```diff
@@ -433,42 +434,54 @@ static int get_if_device_is_in_use(int sequential_id)
...
 
-	for (i = 0; i < num_platforms; i++) {
-		platforms[i].platform = platform_list[i];
+	if (ret != CL_SUCCESS)
+		num_platforms = 0;
 
-		HANDLE_CLERROR(clGetPlatformInfo(platforms[i].platform,
-		                                 CL_PLATFORM_NAME, sizeof(opencl_data), opencl_data, NULL),
-		               "Error querying PLATFORM_NAME");
+	if (num_platforms < 1 && options.verbosity > VERB_LEGACY)
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Not needed on a regular run.

File: src/opencl_common.c [hunk 4]
```diff
@@ -433,42 +434,54 @@ static int get_if_device_is_in_use(int sequential_id)
...
+	ret = clGetPlatformIDs(MAX_PLATFORMS, platform_list, &num_platforms);
 
-	for (i = 0; i < num_platforms; i++) {
-		platforms[i].platform = platform_list[i];
+	if (ret != CL_SUCCESS)
+		num_platforms = 0;
 
-		HANDLE_CLERROR(clGetPlatformInfo(platforms[i].platform,
-		                                 CL_PLATFORM_NAME, sizeof(opencl_data), opencl_data, NULL),
-		               "Error querying PLATFORM_NAME");
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Not needed on a regular run.

File: src/opencl_common.c [hunk 5]
```diff
@@ -433,42 +434,54 @@ static int get_if_device_is_in_use(int sequential_id)
...
-	char opencl_data[LOG_SIZE];
-	cl_uint num_platforms, device_num, device_pos = 0;
-	int i, ret;
+	cl_uint num_platforms, device_pos = 0;
+	int ret, i;
 
 	/* Find OpenCL enabled devices. We ignore error here, in case
 	 * there is no platform and we'd like to run a non-OpenCL format. */
-	clGetPlatformIDs(MAX_PLATFORMS, platform_list, &num_platforms);
+	ret = clGetPlatformIDs(MAX_PLATFORMS, platform_list, &num_platforms);
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Print the error code (if needed) using increased verbosity.

File: src/opencl_common.c [hunk 6]
```diff
@@ -433,42 +434,54 @@ static int get_if_device_is_in_use(int sequential_id)
...
+		                   MAX_GPU_DEVICES - device_pos, /* avoid buffer overrun */
+		                   &devices[device_pos], &num_devices);
+		if (ret != CL_SUCCESS)
+			num_devices = 0;
 
-		if ((ret != CL_SUCCESS || device_num < 1) &&
-		        options.verbosity > VERB_LEGACY)
-			fprintf(stderr, "No OpenCL devices was found on platform #%d\n", i);
+		if (num_devices < 1 && options.verbosity > VERB_LEGACY)
+			fprintf(stderr, "No OpenCL devices were found on platform #%d (%d)\n", i, ret);
```
Comment (claudioandre-br) [inline src/opencl_common.c]: Idem. Print the error code (if needed) using increased verbosity.

**Comment by magnumripper** [comment]
I rebased #3450 upon this one now (also rebased, squashed and force-pushed that branch)

**Pull request closed without merging**